WordPress/Classes/Networking/AccountServiceRemoteREST.m

```diff
@@ -80,14 +80,16 @@ - (RemoteUser *)remoteUserFromDictionary:(NSDictionary *)dictionary
 
 - (NSArray *)remoteBlogsFromJSONArray:(NSArray *)jsonBlogs
 {
-    NSMutableArray *remoteBlogs = [NSMutableArray arrayWithCapacity:[jsonBlogs count]];
-    for (NSDictionary *jsonBlog in jsonBlogs) {
-        BOOL isJetpack = [jsonBlog[@"jetpack"] boolValue];
-        if (!isJetpack || JetpackREST.enabled) {
-            [remoteBlogs addObject:[self remoteBlogFromJSONDictionary:jsonBlog]];
-        }
+    NSArray *blogs = jsonBlogs;
+    if (!JetpackREST.enabled) {
+        blogs = [blogs wp_filter:^BOOL(NSDictionary *jsonBlog) {
+            BOOL isJetpack = [jsonBlog[@"jetpack"] boolValue];
+            return !isJetpack;
+        }];
     }
-    return [NSArray arrayWithArray:remoteBlogs];
+    return [blogs wp_map:^id(NSDictionary *jsonBlog) {
+        return [self remoteUserFromDictionary:jsonBlog];
```

Comment:
This one should be `remoteBlogFromJSONDictionary` (instead of remoteUser). Otherwise it'll crash, violently